substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/xml/JavaxXmlClassAndResourcesLoaderFeature.java

```diff
@@ -36,13 +36,24 @@
 
 import com.oracle.svm.core.feature.InternalFeature;
 import com.oracle.svm.core.jdk.JNIRegistrationUtil;
+import com.oracle.svm.util.ReflectionUtil;
 import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
 
 @AutomaticallyRegisteredFeature
 public class JavaxXmlClassAndResourcesLoaderFeature extends JNIRegistrationUtil implements InternalFeature {
 
     @Override
     public void beforeAnalysis(BeforeAnalysisAccess access) {
+        /*
+         * Ensure that class loading of xml related classes work.
+         * com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl
+         * implicitly depends on jdk.xml.internal.JdkXmlUtils. So initialization
+         * order needs to happen in the reverse order.
+         */
+        Class<?> jdkUtil = access.findClassByName("jdk.xml.internal.JdkXmlUtils");
+        if (jdkUtil != null) {
+            ReflectionUtil.newInstance(jdkUtil);
```

Comment:
Use `RuntimeClassInitialization.initializeAtBuildTime(jdkUtil)` because as far as I understand it you only care that the class gets initialized, there is no need to actually instantiate the class.

Reply:
Unfortunately, that does not work because it would initialize it too late. For the same reason, reachability handlers and `--initialize-at-build-time` does not work.

Reply:
Maybe something like:
```
        ClassInitializationSupport.singleton().forceInitializeHosted(jdkUtil, "break circular class initialization in XML code", false);
```

Reply:
Hm, that seems to be a difference between our old `ProvenSafeClassInitializationSupport` and the new default `AllowAllHostedUsagesClassInitializationSupport`. The old policy immediately initialized the class when it was registered, which is actually a good thing I think - we also have other places where we need to mitigate class initialization deadlocks, and that seems like a simple way to do it.

Reply:
I've used `forceInitializeHosted()` now.

Reply:
@christianwimmer thinks we should investigate changing the `AllowAllHostedUsagesClassInitializationSupport` first. This might open up different opportunities to solve the original problem. I'll give that a try come back to you.

Reply:
With the change suggested by @christianwimmer, one could use `--initialize-at-build-time=jdk.xml.internal.JdkXmlUtils` to mitigate the problem.
@jerboaa would that be acceptable for Quarkus as well?

Reply:
I'll have to check and will get back to you later today or tomorrow. We don't really have a quarkus reproducer, though. So verification will be getting a fix integrated and then rely on many CI runs to see if the problem persists after a fix has been pushed.

Reply:
Ok sounds good. FYI, #8185 is the `AllowAllHostedUsagesClassInitializationSupport` change I'm currently running through our CI, in case you want to have a peek.

Reply:
Quarkus initializes everything at build time (except those few classes that can't be initialized at build time or are causing issues) so `jdk.xml.internal.JdkXmlUtils` as well is registered for build-time initialization (although I expect this to be the case even outside Quarkus since it's a core JDK class).

I have also confirmed this by running one of the Quarkus tests that triggers the issue with `-H:+PrintClassInitialization`.

Having said that I believe the suggested solution would work out of the box for Quarkus.

cc @jerboaa

Reply:
That would be great. So I will continue with #8185 and we see if the issues on your side disappear once it is merged.

Reply:
Sounds good to me.

Reply:
@zapster I'm rebooting this conversation. https://github.com/oracle/graal/pull/8185 or the replacement, https://github.com/oracle/graal/pull/8193 So I'm reopening this PR.